.travis.yml

```diff
@@ -22,51 +27,48 @@ matrix:
         - python: 3.6
           env: KERAS_BACKEND=cntk PYTHONWARNINGS=ignore
 install:
-  # code below is taken from http://conda.pydata.org/docs/travis.html
-  # We do this conditionally because it saves us some downloading if the
-  # version is the same.
-  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
-      wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh;
-    else
-      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
-    fi
-  - bash miniconda.sh -b -p $HOME/miniconda
   - export PATH="$HOME/miniconda/bin:$PATH"
-  - hash -r
+  # Inital setup if cache dir does not exist
+  - if [[ ! -d $HOME/miniconda/bin ]]; then
+      if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
+        wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh;
+      else
+        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+      fi;
+      bash miniconda.sh -f -b -p $HOME/miniconda;
+      conda config --set always_yes yes --set changeps1 no;
+      conda create --copy -q -n test-environment python=$TRAVIS_PYTHON_VERSION;
+    fi
   - conda config --set always_yes yes --set changeps1 no
+  - hash -r
+  -
   - conda update -q conda
-  # Useful for debugging any issues with conda
-  - conda info -a
-
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy nose scipy matplotlib pandas pytest h5py
   - source activate test-environment
-  - conda install mkl mkl-service
-  - pip install theano
-
-  # set library path
-  - export LD_LIBRARY_PATH=$HOME/miniconda/envs/test-environment/lib/:$LD_LIBRARY_PATH
-
+  -
+  # Collect required packages to minimize calls to conda
   # install PIL for preprocessing tests
   - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
-      conda install pil;
+      export ADDITIONAL_CONDA_PACKAGES="pil";
     elif [[ "$TRAVIS_PYTHON_VERSION" == "3.6" ]]; then
-      conda install Pillow;
+      export ADDITIONAL_CONDA_PACKAGES="Pillow";
     fi
+  -
```

Comment:
Why the empty lines with `-`?

Reply:
The empty lines were already previously here as whitespace to separate different blocks.